cabal-install/Distribution/Client/List.hs

```diff
@@ -106,11 +106,11 @@ getPkgList verbosity packageDBs repoCtxt comp progdb listFlags pats = do
                         (         PackageIndex.allPackages sourcePkgIndex)
 
             -- gather info for packages matching search term
-          | otherwise = pkgsInfoMatching
+          | otherwise = pkgsInfoMatching exactMatch
 
         pkgsInfoMatching ::
-          [(PackageName, [Installed.InstalledPackageInfo], [UnresolvedSourcePackage])]
-        pkgsInfoMatching =
+          Bool -> [(PackageName, [Installed.InstalledPackageInfo], [UnresolvedSourcePackage])]
```

Comment:
You don't need to pass `Bool` here. `exactMatch` would be visible as the `where` is shared across all of `getPkgList`